python-lib/text_preprocessing.py

```
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+from typing import List, AnyStr
+import re
+import spacy.lang
+import pandas as pd
+import logging
+import string
+from plugin_io_utils import generate_unique
+
+
+from language_dict import SUPPORTED_LANGUAGES
+
+class TextPreprocessor:
+    
+    PUNCTUATION = "!\"#$%&()*+,-./:;<=>?@[\\]^_`{|}~_！？｡。＂＃＄％＆＇（）＊＋，－／：；＜＝＞＠［＼］＾＿｀｛｜｝～｟｠｢｣､、〃《》「」『』【】〔〕〖〗〘〙〚〛〜〝〞〟〰〾〿–—‘’‛“”„‟…‧﹏."
+    
+    def __init__(self):
+        self.tokenizers = {}
+        self.supported_lang_code = SUPPORTED_LANGUAGES.keys()
+        
+    def _add_tokenizers(self, lang_code_new_list: List[AnyStr]):
+        
+        language_modules = {}
+        nlps = {}
+        
+        for lang_code in lang_code_new_list:
+            
+            if lang_code != lang_code: # check for NaNs
```

Not the most understandable solution to check for NaNs, I would rather test that len(lang_code) is 2

---

I thought None could be given by some DB, and None has no length. Do you infirm this ?

---

I am not sure. How about we "validate" language_code with `lang_code in SUPPORTED_LANGUAGE_CODE`, else log a warning. That would cover all invalid cases.

---

Not sure to understand: you want to raise only one warning each time an encontered lang code is unsupported ?

what would be the benefit to add in a spacy nlp object without specific language ?

---

Yes, one warning per unsupported language code.
spacy nlp has a default tokenizer which is not language specific. This tokenization library should be as generic and reusable as possible, so have this default tokenizer as a fallback is a good thing.